app/projects/public/hsrc/executables/DualNBack.service.js

```diff
@@ -0,0 +1,274 @@
+'use strict';
+
+tatool
+  .factory('DualNBack', [ 'executableUtils', 'dbUtils', 'timerUtils', 'stimulusServiceFactory', 'inputServiceFactory',
+    function (executableUtils, dbUtils, timerUtils, stimulusServiceFactory, inputServiceFactory) {
+
+    var DualNBack = executableUtils.createExecutable();
+
+    var DISPLAY_DURATION_DEFAULT = 2000;
+    var N_VALUE = 1;
+    var ITERATIONS = 20;
+    var TARGETS = 12;
+
+    //  Initialze variables at the start of every session
+    DualNBack.prototype.init = function() {
+      var deferred = executableUtils.createPromise();
+
+      if (!this.showKeys) {
+        this.showKeys = { propertyValue: true };
+      } else {
+        this.showKeys.propertyValue = (this.showKeys.propertyValue === true) ? true : false;
+      }
+
+      if (!this.timerEnabled) {
+        this.timerEnabled = { propertyValue: false };
+      } else {
+        this.timerEnabled.propertyValue = (this.timerEnabled.propertyValue === true) ? true : false;
+      }
+      
+      if (!this.stimuliPath) {
+        deferred.reject('Invalid property settings for Executable DualNBack. Expected property <b>stimuliPath</b> of type Path.');
+      }
+
+      // template properties
+      this.stimulusService = stimulusServiceFactory.createService(this.stimuliPath);
```

Comment:
This is not a required change but I thought I'd mention it here as this task might benefit from it. In case a task doesn't have an inter stimulus interval and one wants to avoid the flickering of hiding/showing a stimulus, we introduced a 2nd parameter for the createService method of stimulus, input, grid: defaultVisible = false. By setting this to true the template service will be visible by default and by removing any hide() calls in your controller you should be able to get rid of the flickering between trials.

See doc on createService method:
https://www.tatool-web.com/#/doc/ref-directive-tatoolStimulus.html

Reply:
Sounds good to me.